File: scraping-service/models/scrape.go
package models

import "time"

type ScrapeRequest struct {
	URL          string `json:"url"`
	UserID       string `json:"user_id"`
	AnalysisType string `json:"analysis_type"`
	CustomScript string `json:"custom_script,omitempty"`
}

type ScrapeResult struct {
	OrderID      string    `json:"order_id"`
	UserID       string    `json:"user_id"`
	CreatedAt    time.Time `json:"created_at"`
	URL          string    `json:"url"`
	AnalysisType string    `json:"analysis_type"`
	CustomScript string    `json:"custom_script,omitempty"`
	Data         string    `json:"data"`
}

File: admin-service/src/models/Scrape.js
//ScrapeSmith\admin-service\src\models\Scrape.js

import mongoose from 'mongoose';

const scrapeSchema = new mongoose.Schema({
  order_id:      { type: String, required: true },
  user_id:       { type: String, required: true },
  created_at:    { type: Date,   default: Date.now },
  url:           { type: String, required: true },
  analysis_type: { type: String, required: true },
  custom_script: { type: String },
  data:          { type: String, required: true },
}, { collection: 'scraped_data' });

export const Scrape = mongoose.model('Scrape', scrapeSchema);
